src/test/scala/chiselTests/BundleElementsSpec.scala

```diff
@@ -0,0 +1,457 @@
+        val a = UInt(8.W)
+        val b = Bool()
+        val c = Enum0.Type
+      }))
+      assert(FieldsMatch(out, "b", "a"))
+    })
+  }
+
+  "plugin should error correctly when bundles contain only a Option field" in {
+    ChiselStage.emitFirrtl(new Module {
+      val io = IO(new Bundle {
+        val foo = Input(UInt(8.W))
+        val x = new Bundle {
+          val y = if (false) Some(Input(UInt(8.W))) else None
+        }
+      })
+      assert(FieldsMatch(io, "x", "foo"))
+      assert(FieldsMatch(io.x))
+    })
+  }
+
+  "plugin should handle fields using the boolean to option construct" in {
+    case class ALUConfig(
+      xLen: Int,
+      mul:  Boolean,
+      b:    Boolean)
+
+    class OptionalBundle extends Bundle {
+      val optionBundleA = Input(UInt(3.W))
+      val optionBundleB = Input(UInt(7.W))
+    }
+
+    class ALU(c: ALUConfig) extends Module {
+
+      class BpipOptionBundle extends Bundle with IgnoreSeqInBundle {
+        val bpipUIntVal = Input(UInt(8.W))
+        lazy val bpipUIntLazyVal = Input(UInt(8.W))
+        var bpipUIntVar = Input(UInt(8.W))
+        def bpipUIntDef = Input(UInt(8.W))
+
+        val bpipOptionUInt = Some(Input(UInt(16.W)))
+        val bpipOptionOfBundle = if (c.b) Some(new OptionalBundle) else None
+        val bpipSeqData = Seq(UInt(8.W), UInt(8.W))
+      }
+
+      val io = IO(new BpipOptionBundle)
+      assert(FieldsMatch(io, "bpipUIntLazyVal", "bpipOptionUInt", "bpipUIntVar", "bpipUIntVal"))
+    }
+
+    ChiselStage.emitFirrtl(new ALU(ALUConfig(10, mul = true, b = false)))
+  }
+
+  "TraceSpec test, different version found in TraceSpec.scala" in {
+    class Bundle0 extends Bundle {
+      val a = UInt(8.W)
+      val b = Bool()
+      val c = Enum0.Type
+    }
+
+    class Bundle1 extends Bundle {
+      val a = new Bundle0
+      val b = Vec(4, Vec(4, Bool()))
+    }
+
+    class Module0 extends Module {
+      val i = IO(Input(new Bundle1))
+      val o = IO(Output(new Bundle1))
+      val r = Reg(new Bundle1)
+      o := r
+      r := i
+
+      assert(FieldsMatch(i, "b", "a"))
+      assert(FieldsMatch(o, "b", "a"))
+      assert(FieldsMatch(r, "b", "a"))
+    }
+
+    class Module1 extends Module {
+      val i = IO(Input(new Bundle1))
+      val m0 = Module(new Module0)
+      m0.i := i
+      m0.o := DontCare
+      assert(FieldsMatch(i, "b", "a"))
+
+    }
+
+    object Enum0 extends ChiselEnum {
+      val s0, s1, s2 = Value
+    }
+
+    ChiselStage.emitFirrtl(new Module1)
+  }
+}
+
+object FieldsMatch {
+  def apply(bundle: Bundle, fieldNames: String*): Boolean = {
+    fieldNames.toSeq
+      .zipAll(bundle.elements.map(_._1).toList, "Missing Bundle Field", "Extra BundleField")
+      .forall {
+        case (fromBundle, fromList) =>
+          if (fromBundle != fromList) {
+            println(s"""Bundle: ${bundle}, field mismatch "$fromBundle" to "$fromList" """)
+          }
+          fromBundle == fromList
+      }
+  }
```

Comment:
I really would prefer if this matched just not names but also the actual Data themselves. I think a utility like the following would be slightly more verbose for each check but actually meaningfully checks that we aren't accidentally cloning the Data or something.
```scala
def checkBundle[T <: Bundle](bun: T)((checks: T => (String, Data))*): Unit = {
  bun.elements.toSeq should equal checks.map(bun).toSeq
}

checkBundle(new BpipOneField)(
  "bpipSuperTraitGood" -> _.bpipSuperTraitGood,
  "bpipTraitGood" -> _.bpipTraitGood,
  ...
)```

Reply:
done, I have pushed this commit into PR